src/track/serato/markers2.cpp

```diff
@@ -438,6 +438,34 @@ QByteArray SeratoMarkers2::dump() const {
     return outerData.leftJustified(size, '\0');
 }
 
+QList<CueInfo> SeratoMarkers2::getCues(double timingOffsetMillis) const {
+    qDebug() << "Reading cues from 'Serato Markers2' tag data...";
+    QList<CueInfo> cueInfos;
+    for (auto& pEntry : m_entries) {
+        DEBUG_ASSERT(pEntry);
+        switch (pEntry->typeId()) {
+        case SeratoMarkers2Entry::TypeId::Cue: {
+            const SeratoMarkers2CueEntry* pCueEntry = static_cast<SeratoMarkers2CueEntry*>(pEntry.get());
+            CueInfo cueInfo(
+                    CueType::HotCue,
+                    pCueEntry->getPosition() + timingOffsetMillis,
+                    std::nullopt,
+                    pCueEntry->getIndex(),
+                    pCueEntry->getLabel(),
+                    pCueEntry->getColor());
+            cueInfos.append(cueInfo);
+
+            break;
+        }
+        // TODO: Add support for COLOR/LOOP/FLIP
```

Comment:
I thought track color support was added in #2499. I don't think it would be much work to import the first found loop here. Of course we can't do anything useful with the Flip data yet.

Reply:
> I thought track color support was added in #2499. It looks like the cue color is handled above. So I do not understand the TODO for COLOR.

Correct, the comment is outdated.

> I don't think it would be much work to import the first found loop here. Of course we can't do anything useful with the Flip data yet.

Thinking about it, we could even import all loops. I think Mixxx will just ignore the additional ones, but that might save users from needing to reimport the metadata when saved loop support lands in 2.4.

Reply:
If Mixxx ignores additional loops, yeah let's import them all. We could do the same for Rekordbox too.

Reply:
On second thought it might be more complicated. Mixxx shares hotcue numbers between loops and cues, Serato does not. We could import loops with offset 16 or something like that though.

Reply:
We also don't have support for locked cues yes, so we have to ignore that attribute.

Reply:
> Mixxx shares hotcue numbers between loops and cues

Huh? Mixxx only supports one loop currently; it has no number.

Reply:
Yes, but #2194 currently implements a context-sensitive way to set cues/loops for controllers without a dedicated loop mode. So hotcue 1 could be a cue, hotcue 2 a loop, and hotuce 3 another cue. Serato has separate lists for cues and loops.

Reply:
> We could import loops with offset 16 or something like that though.

If we did that, after #2194 is merged, users would have to have >16 hotcues showing in the skin to see the imported loops. I propose setting the numbers for the imported loops starting from the highest number of the imported hotcues.

Reply:
Hmm, that might be useful to some users, but my controller has dedicated loop and hotcue modes. I want all my regular cues show up in hotcue mode and my loops show up in loop mode. Otherwise the labels on my controller make no sense and I will get confused during sets what which button is a loop and which is hotcue. I think this should be made configurable. Anyway, I would rather do this in a separate follow-up PR and just import the first loop for now to keep this PR at a reasonable size.

Reply:
How about importing loops starting with a hotcue number of 8? AFAIK no Serato controllers have more than 8 buttons for hotcues/loops. We only show 8 hotcue buttons on screen in Mixxx, but the loops would still show on the waveforms and be usable on a controller. Later we may consider adding some way to toggle between pages hotcue buttons to somewhat mimic Serato.

Reply:
ping

Reply:
I'm okay with that. I just google'd a bit and found this in the [Serato DJ Pro manual](https://support.serato.com/hc/en-us/articles/226518228-Cue-Points):

> Serato DJ has up to 8 Cue Points

So this should work. I'll look into it tomorrow.